app/components-react/sidebar/SideNav.tsx

```diff
@@ -1,255 +1,412 @@
+
   return (
-    <div
-      onMouseEnter={() => setShowTabDropdown(true)}
-      onMouseLeave={() => setShowTabDropdown(false)}
-    >
-      <div
-        className={cx(styles.primaryTab, {
-          [styles.active]: p.currentPage === 'Studio' && currentTab === 'default',
-        })}
+    <Layout hasSider className="sidenav">
+      <Sider
+        collapsible
+        collapsed={!isOpen}
+        trigger={null}
+        className={cx(
+          styles.sidenav,
+          styles.sidenavSider,
+          !isOpen && styles.siderClosed,
+          !leftDock && styles.noLeftDock,
+        )}
+        ref={sider}
       >
-        <StudioTab page={studioTabs[0]} navigate={p.navigate} />
-        {studioTabs.length > 1 && (
-          <i
-            className={cx('icon-down', styles.studioDropdown, {
-              [styles.studioDropdownActive]: currentTab !== 'default',
+        <Scrollable className={cx(styles.sidenav, styles.sidenavScroll)}>
+          <Menu
+            key={ENavName.TopNav}
+            forceSubMenuRender
+            mode="inline"
+            className={cx(
+              styles.menuContainer,
+              isOpen && styles.open,
+              !isOpen && (styles.siderClosed, styles.menuWrapper),
+            )}
+            defaultOpenKeys={openMenuItems && openMenuItems}
+            defaultSelectedKeys={[EMenuItem.Editor]}
+          >
+            {menuItems.map((menuItem: IParentMenuItem) => {
+              if (
+                !menuItem?.isActive ||
+                (menuItem.title === EMenuItem.ThemeAudit && !themeAuditEnabled)
+              ) {
+                // skip inactive menu items
+                // skip legacy menu items for new users
+                // skip Theme Audit if not enabled
+                return null;
+              } else if (menuItem.title === EMenuItem.Editor && loggedIn && studioTabs.length > 1) {
+                // if closed, show editor tabs in sidenav
+                // which can be toggled to show or hide
+                // otherwise, show editor tabs in submenu
+                // don't translate tab title because the user has set it
+                return showCustomEditor && !isOpen ? (
+                  studioTabs.map(tab => (
+                    <Menu.Item
+                      key={tab.key}
+                      className={cx(
+                        styles.sidenavItem,
+                        !isOpen && styles.closed,
+                        currentMenuItem === tab.key && styles.active,
+                      )}
+                      title={tab.title}
+                      icon={<i className={tab.icon} />}
+                      onClick={() => navigateToStudioTab(tab.target, tab.trackingTarget, tab.key)}
+                    >
+                      {tab.title}
+                    </Menu.Item>
+                  ))
+                ) : (
+                  <Menu.SubMenu
+                    key={menuItem.key}
+                    title={$t(menuItem.title)}
```

Comment:
ah i see how the title translation is happening now... i guess that works but probably not a pattern we want to adopt widely 🤔

Reply:
Closing because I refactored to use the menu title map